extensions/interactions/ImageClickInput/ImageClickInput.py

```diff
@@ -68,3 +70,211 @@ class ImageClickInput(base.BaseInteraction):
         # explicitly addressed.
         'addressed_info_is_supported': False,
     }]
+
+    @classmethod
+    def to_proto(
+        cls, default_outcome, customization_args, hints, answer_groups
+    ):
+        """Creates a ImageClickInputInstance proto object.
+
+        Args:
+            default_outcome: Outcome. The domain object.
+            customization_args: CustominzationArgs. The domain object.
+            hints: Hint. The domain object.
+            answer_groups: AnswerGroups. The domain object.
+
+        Returns:
+            ImageClickInputInstance. The proto object.
+        """
+        customization_args_proto = (
+            cls._to_customization_args_proto(
+                customization_args)
+        )
+        outcome_proto = default_outcome.to_proto()
+        hints_proto_list = cls.get_hint_proto(cls, hints)
+        answer_groups_proto = cls._to_answer_groups_proto(
+            answer_groups)
+
+        return state_pb2.ImageClickInputInstance(
+            customization_args=customization_args_proto,
+            default_outcome=outcome_proto,
+            hints=hints_proto_list,
+            answer_groups=answer_groups_proto
+        )
+
+    @classmethod
+    def _to_answer_groups_proto(cls, answer_groups):
+        """Creates a AnswerGroup proto object
+            for ImageClickInputInstance.
+
+        Args:
+            answer_groups: list(AnswerGroup). List of answer groups of the
+                interaction instance.
+
+        Returns:
+            list. The AnswerGroup proto object list.
+        """
+        answer_group_list_proto = []
+
+        for answer_group in answer_groups:
+            base_answer_group_proto = answer_group.to_proto()
+            rules_spec_proto = cls._to_rule_specs_proto(
+                answer_group.rule_specs)
+            answer_group_proto = state_pb2.ImageClickInputInstance.AnswerGroup(
+                base_answer_group=base_answer_group_proto,
+                rule_specs=rules_spec_proto
+            )
+            answer_group_list_proto.append(answer_group_proto)
+
+        return answer_group_list_proto
+
+    @classmethod
+    def _to_rule_specs_proto(cls, rule_specs_list):
+        """Creates a RuleSpec proto object list.
+
+        Args:
+            rule_specs_list: list(RuleSpec). List of rule specifications.
+
+        Returns:
+            list. The RuleSpec proto object list.
+        """
+        rule_specs_list_proto = []
+        rules_specs_proto = {}
+
+        for rule_spec in rule_specs_list:
+            rule_type = rule_spec.rule_type
+            if rule_type == 'IsInRegion':
+                is_in_region_proto = cls._to_is_in_image_region_proto(
+                    rule_spec.inputs['x'])
+                rules_specs_proto = state_pb2.ImageClickInputInstance.RuleSpec(
+                    is_in_region=is_in_region_proto)
+
+            rule_specs_list_proto.append(rules_specs_proto)
+
+        return rule_specs_list_proto
```

Comment:
I'm a bit confused by this, why do we only accept rules of type `'IsInRegion'`?

Reply:
Fixed, as of now, we only have one rule in image.